rules_gapic/bazel/src/main/java/com/google/api/codegen/bazel/BazelBuildFileTemplate.java

```diff
@@ -32,7 +38,71 @@ private String expand(Map<String, String> tokens) {
     return builder.toString();
   }
 
-  String expand(BazelBuildFileView bpv) {
-    return this.expand(bpv.getTokens());
+  String expand(BazelBuildFileView bpv) throws IOException {
+    String expandedTemplate = this.expand(bpv.getTokens());
+
+    // Apply overrides
+    Map<String, Map<String, String>> overriddenStringAttributes =
+        bpv.getOverriddenStringAttributes();
+    Map<String, Map<String, List<String>>> overriddenListAttributes =
+        bpv.getOverriddenListAttributes();
+    Map<String, String> assemblyPkgRulesNames = bpv.getAssemblyPkgRulesNames();
+    if (overriddenStringAttributes.size() == 0
+        && overriddenListAttributes.size() == 0
+        && assemblyPkgRulesNames.size() == 0) {
+      // nothing to override
+      return expandedTemplate;
+    }
+
+    // write the content of the build file to a temporary directory and fix it with Buildozer
+    File tempdir = Files.createTempDirectory("build_file_generator_").toFile();
+    File buildBazel = new File(tempdir, "BUILD.bazel");
+    Path buildBazelPath = buildBazel.toPath();
+    Files.write(buildBazelPath, expandedTemplate.getBytes(StandardCharsets.UTF_8));
+
+    Buildozer buildozer = Buildozer.getInstance();
+
+    // First of all, rename the rules
+    for (Map.Entry<String, String> entry : assemblyPkgRulesNames.entrySet()) {
+      String kind = entry.getKey();
+      String newName = entry.getValue();
+      // Note: if there are multiple *_gapic_assembly_pkg rules for the same language,
```

Comment:
So... for now I still think we should make it opt-in since regeneration with preserving parameters (e.g. for TypeScript generator) is going to be more common use case.

Reply:
I removed this comment because I'm now throwing an exception in this case during initial parsing.